go-controller/pkg/ovn/egressgw.go

```diff
@@ -485,26 +485,24 @@ func (oc *Controller) deleteGWRoutesForPod(name ktypes.NamespacedName, podIPNets
 				node := util.GetWorkerFromGatewayRouter(gr)
 				portPrefix, err := oc.extSwitchPrefix(node)
 				if err != nil {
-					klog.Infof("Failed to find ext switch prefix for %s %v", node, err)
-					continue
+					return fmt.Errorf("failed to find ext switch prefix for %s %v", node, err)
 				}
 				if err := oc.deleteLogicalRouterStaticRoute(pod, mask, gw, gr); err != nil {
-					klog.Errorf("Unable to delete ECMP route for pod: %s to GR %s, GW: %s, err:%v",
+					return fmt.Errorf("unable to delete ECMP route for pod: %s to GR %s, GW: %s, err:%v",
 						name, gr, gw, err)
-				} else {
-					delete(routeInfo.podExternalRoutes[pod], gw)
-					klog.V(5).Infof("ECMP route deleted for pod: %s, on gr: %s, to gw: %s", name,
-						gr, gw)
 				}
+				delete(routeInfo.podExternalRoutes[pod], gw)
+				klog.V(5).Infof("ECMP route deleted for pod: %s, on gr: %s, to gw: %s", name, gr, gw)
 				if entry := routeInfo.podExternalRoutes[pod]; len(entry) == 0 {
 					if err := oc.delHybridRoutePolicyForPod(podIPNet.IP, node); err != nil {
```

Comment:
lgtm. I tried to convince myself that all these calls are idem-potent enough such that is only a subset of them succeeded, it would be okay to visit the entire function over again and no new errors would occur due to it being partially finished. I think I did convince myself. :)